amplfi/train/models/flow.py

```diff
@@ -124,6 +124,69 @@ def on_test_epoch_start(self):
         self.test_results: list[Result] = []
         self.idx = 0
 
+    def _get_log_prior_dict(self, trainer=None):
+        """
+        Get the log prior for each parameter in the model.
+        """
+        if trainer is not None:
+            self.trainer = trainer
```

Comment:
This step should not be necessary. (I know we talked about this; mentioning here for my own sake).